### ISSUE TITLE
Provide a more convenient way to set an exception as "active", from Python code

### ISSUE THREAD

**Issue by njsmith** [python/cpython#89524]
BPO | [45361](https://bugs.python.org/issue45361)
--- | :---
Nosy | @gvanrossum, @ncoghlan, @njsmith, @markshannon, @1st1, @iritkatriel

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-10-04.06:42:56.085>
labels = ['type-feature']
title = 'Provide a more convenient way to set an exception as "active", from Python code'
updated_at = <Date 2021-10-10.05:47:09.812>
user = 'https://github.com/njsmith'
```

bugs.python.org fields:
```python
activity = <Date 2021-10-10.05:47:09.812>
actor = 'gvanrossum'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2021-10-04.06:42:56.085>
creator = 'njs'
dependencies = []
files = []
hgrepos = []
issue_num = 45361
keywords = []
message_count = 4.0
messages = ['403121', '403122', '403287', '403572']
nosy_count = 6.0
nosy_names = ['gvanrossum', 'ncoghlan', 'njs', 'Mark.Shannon', 'yselivanov', 'iritkatriel']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue45361'
versions = []
```

</p></details>

**Comment by iritkatriel** [comment]
> In a brief off-line chat we concluded that the desired API would be closer to PyErr_Restore().

It doesn't make sense to me to call PyErr_Restore() from python without raising an exception. What does that mean?

**Comment by gvanrossum** [comment]
I suppose you pass it an exception you caught and saved earlier?

**Comment by gvanrossum** [comment]
I don't think the proposal was to reuse PyErr_Restore(). I think the point was that it should look similar to that.

But honestly unless @njsmith responds this issue isn't going to be resolved for 3.11, it just doesn't seem to have priority (I can't find the time to really read Nathaniel's message with use cases, sorry).

**Comment by iritkatriel** [comment]
The original post is talking about the exc_info (caught exception). Later (offline) it became ‘something like’ pyerr_restore (in-flight exception). It’s totally confused, and we’re not getting and clarifications from the OP, so I am closing this.
